### PR TITLE
Vg 46 fix product images not going through at all to any branch (UPDATED API)

### PR DESCRIPTION
**Pull request by Danback**
I've updated our API, new one is "https://fakestoreapi.com/". It is cleaner, easier to handle and more effective.

Added in this branch:
* Changed API to make products etc work
* Added styling etc to said products
* minor changes to productview component etc

### CODE REVIEW

**Review by Demppas — APPROVED** [review]
ser bra ut!

File: src/components/ProductCard.vue [hunk 1]
```diff
@@ -1,12 +1,59 @@
...
+const props = defineProps({
+  product: {
+    type: Object,
+    required: true,
+    default: () => ({})
+  }
+});
 </script>
+
+<style scoped>
```
Comment (marcusju23) [inline src/components/ProductCard.vue:24]: No Tailwind CSS? 🤔

File: src/components/ProductCard.vue [hunk 2]
```diff
@@ -1,12 +1,59 @@
 <template>
   <div class="product-card">
-    <img :src="product.images[0]" alt="product image" class="product-image"/>
-    <h3>{{ product.title }}</h3>
-    <p>{{ product.price }} $</p>
-    <router-link :to="`/product/${product.id}`">View Details</router-link>
+    <a :href="`/product/${product.id}`" class="product-link">
+      <img :src="product.image" alt="product image" class="product-image" />
```
Comment (marcusju23) [inline src/components/ProductCard.vue:4]: "image" is redundant in `alt="product image"`

File: src/router/index.js [hunk 3]
```diff
@@ -26,7 +26,7 @@ const router = createRouter({
         {
             path: "/product/:id",
             name: "ProductDetail",
-            component: ProductDetail
+            component: ProductDetailView
```
Comment (marcusju23) [inline src/router/index.js:29]: Good find 👍

File: src/views/HomeView.vue [hunk 4]
```diff
@@ -1,25 +1,32 @@
 <template>
-  <div class="product-list">
-    <ProductCard v-for="product in products" :key="product.id" :product="product" />
+  <div>
+    <ProductCard
```
Comment (marcusju23) [inline src/views/HomeView.vue:3]: This can be as one line.

File: src/views/ProductDetailView.vue [hunk 5]
```diff
@@ -1,26 +1,76 @@
 <template>
   <div v-if="product">
     <h1>{{ product.title }}</h1>
-    <img :src="product.images?.[0]" alt="product image" v-if="product.images?.length" />
+    <img :src="product.image" alt="product image" />
```
Comment (marcusju23) [inline src/views/ProductDetailView.vue:4]: "image" is redundant in `alt="product image"`

File: src/views/ProductDetailView.vue [hunk 6]
```diff
@@ -1,26 +1,76 @@
...
+
+    <p><strong>Category:</strong> {{ product.category }}</p>
```
Comment (marcusju23) [inline src/views/ProductDetailView.vue:6]: `<strong>` tags can be replaced with Tailwind CSS instead.

File: src/views/ProductDetailView.vue [hunk 7]
```diff
@@ -1,26 +1,76 @@
...
+      error.value = true;
+    }
+  } catch (err) {
+    console.error('Error fetching product details:', err);
+    error.value = true;
   }
 });
 </script>
+
+<style scoped>
```
Comment (marcusju23) [inline src/views/ProductDetailView.vue:47]: No Tailwind CSS? 🤔

**Review by Rishad11 — APPROVED** [review]
Snyggt jobbat!

**Review by marcusju23 — CHANGES_REQUESTED** [review]
Put the styling in the CSS file instead and use Tailwind CSS as it's a project requirement and follows our structure. Other than that, great job! 👍

**Pull request merged**